tests/periph_spi/tests/04__periph_spi_timer.robot

```diff
@@ -0,0 +1,78 @@
+*** Settings ***
+Documentation       Verify that the meassured times are within a certain window
+
+Suite Setup         Run Keywords    PHILIP Reset
+...                                 RIOT Reset
+...                                 API Sync Shell
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    PHILIP Reset
+...                                 RIOT Reset
+...                                 API Sync Shell
+...                                 SPI Init Should Succeed
+Test Teardown       Run Keywords    SPI Release Should Succeed
+
+Resource            periph_spi.keywords.txt
+Resource            api_shell.keywords.txt
+
+Variables           test_vars.py
+
+Force Tags          periph  spi
+
+
+*** Keywords ***
+SPI Clock Speed Check
+    [Documentation]     Main routine to verify the clock speed
+    [Arguments]         ${clk_speed_string}  ${clk_speed_value}  ${size}
+    PHiLIP.write and execute            spi.mode.if_type    3
+    SPI Acquire Should Succeed          0  ${clk_speed_string}
+    SPI Transfer Bytes Should Succeed   cont=0   in_len=${size}
+    API Call Should Succeed             PHiLIP.Read Reg         sys.sys_clk
+    ${sys_clk} =  Set Variable          ${RESULT['data']}
+    API Call Should Succeed             PHiLIP.Read Reg         spi.byte_ticks
+    ${spi_speed_limits} =  Set Variable  
+    ...  ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
+    Should Be True  
+    ...  ${spi_speed_limits}[0] < ${${RESULT['data']} * ${size}} and ${spi_speed_limits}[1] > ${${RESULT['data']} * ${size}}
+    ...  Number of ticks should be between ${spi_speed_limits}[0] and ${spi_speed_limits}[1] but was ${${RESULT['data']} * ${size}}
```

Comment:
Maybe you can divide by the `${sys_clk}`  so we can get  time instead of ticks.

Reply:
```suggestion
    ...  Time should be between ${${spi_speed_limits}[0]/${sys_clk}} and${${spi_speed_limits}[1]/${sys_clk}} but was ${${RESULT['data']} * ${size} / ${sys_clk}}
```

Reply:
Is done. I also advanced it a bit, so that the output is directly in microseconds.